### ISSUE TITLE
prop-types warns even if you don’t use props at all

### ISSUE THREAD

**Issue by lo1tuma** [yannickcr/eslint-plugin-react#4]
The `prop-types` rule warns always if you don’t specify the `propTypes` property within a react component even if you don’t use any `props`.

I suggest to only warn if a used prop is not defined within `propTypes`. This would also be good if you have specified the `propTypes` property somehow but not defined every used prop.

So the following should not be considered as a warning:

```jsx
var Hello = React.createClass({
  render: function() {
    return <div>Hello World</div>;
  }
});
```

But this should be considered as a warning:

```jsx
var Hello = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired
  },
  render: function() {
    return <div>Hello {this.props.name} and {this.props.propWithoutTypeDefinition}</div>;
  }
});

```

**Comment by yannickcr** [comment]
You are right, warn only if a used prop is not defined within `propTypes` seems to be the way to go.
